commands/provider/service_provider.go

```diff
@@ -103,6 +103,12 @@ func (p *ServiceProvider) ServiceListenEvents(id, eventFilter string) (chan *cor
 			}
 		}
 	}()
+
+	// wait for server to notify the stream is ready.
+	if _, err := stream.Header(); err != nil {
```

Comment:
There is not check if we recive metadata with status ready. It's error prone if someone add some metadata header in the future (send before this md), he will break this part.

Reply:
I wanted to keep the logic on the client side very simple.

If it need to check the header and its value, then it also need to get the right header (so adding a for around?).

Reply:
But for now it even dosen't check values, so there is a branch you have created and I thought about similar funciton, maybe it could be placed inside the `header` package.

Reply:
Done